<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>PlantManagementWidget</class>
 <widget class="QWidget" name="PlantManagementWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>900</width>
    <height>600</height>
   </rect>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QGroupBox" name="addPlantGroup">
     <property name="title">
      <string>Add New Plant</string>
     </property>
     <layout class="QHBoxLayout" name="addPlantLayout">
      <item>
       <widget class="QLabel" name="plantTypeLabel">
        <property name="text">
         <string>Plant Type:</string>
        </property>
       </widget>
      </item>
            <item>
             <widget class="QComboBox" name="plantTypeCombo">
                <property name="sizeAdjustPolicy">
                 <enum>QComboBox::AdjustToContents</enum>
                </property>
                <property name="minimumContentsLength">
                 <number>12</number>
                </property>
        <item>
         <property name="text">
          <string>Rose</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Sunflower</string>
         </property>
        </item>
        <item>
         <property name="text">
          <string>Cactus</string>
         </property>
        </item>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="addPlantButton">
        <property name="text">
         <string>ADD PLANT</string>
        </property>
       </widget>
      </item>
      <item>
       <spacer name="addSpacer">
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </spacer>
      </item>
      <item>
       <widget class="QPushButton" name="removePlantButton">
        <property name="text">
         <string>REMOVE PLANT</string>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QPushButton" name="refreshButton">
        <property name="text">
         <string>REFRESH</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="plantListGroup">
     <property name="title">
      <string>Plant Inventory</string>
     </property>
     <layout class="QVBoxLayout" name="plantListLayout">
      <item>
       <widget class="QTableWidget" name="plantTable">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>250</height>
         </size>
        </property>
        <property name="selectionBehavior">
         <enum>QAbstractItemView::SelectRows</enum>
        </property>
        <property name="selectionMode">
         <enum>QAbstractItemView::SingleSelection</enum>
        </property>
    <column>
     <property name="text">
      <string>ID</string>
     </property>
    </column>
        <column>
         <property name="text">
          <string>Type</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Growth State</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Health</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Water</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Fertilizer</string>
         </property>
        </column>
        <column>
         <property name="text">
          <string>Price</string>
         </property>
        </column>
       </widget>
      </item>
      <item>
       <widget class="QLabel" name="plantCountLabel">
        <property name="text">
         <string>Total Plants: 0</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="plantDetailsGroup">
     <property name="title">
      <string>Selected Plant Details</string>
     </property>
     <layout class="QVBoxLayout" name="detailsLayout">
      <item>
       <widget class="QPlainTextEdit" name="plantDetailsText">
        <property name="readOnly">
         <bool>true</bool>
        </property>
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>80</height>
         </size>
        </property>
        <property name="placeholderText">
         <string>Select a plant to view details...</string>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <layout class="QHBoxLayout" name="systemLayout">
     <item>
      <widget class="QGroupBox" name="timeSimulationGroup">
       <property name="title">
        <string>Time Simulation</string>
       </property>
       <layout class="QVBoxLayout" name="timeSimLayout">
        <item>
         <widget class="QPushButton" name="timeElapsedButton">
          <property name="text">
           <string>SIMULATE HOUR PASSED</string>
          </property>
          <property name="toolTip">
           <string>Updates all plants and queues maintenance commands</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="simulationInfoLabel">
          <property name="text">
           <string>This updates plant states and creates
maintenance commands in the queue.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
     <item>
      <widget class="QGroupBox" name="queueManagementGroup">
       <property name="title">
        <string>Queue Processing</string>
       </property>
       <layout class="QVBoxLayout" name="queueMgmtLayout">
        <item>
         <widget class="QPushButton" name="processQueueButton">
          <property name="text">
           <string>PROCESS COMMAND QUEUE</string>
          </property>
          <property name="toolTip">
           <string>Executes all queued commands through handler chain</string>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QLabel" name="processingInfoLabel">
          <property name="text">
           <string>This executes queued commands and
updates plant states accordingly.</string>
          </property>
          <property name="wordWrap">
           <bool>true</bool>
          </property>
          <property name="alignment">
           <set>Qt::AlignCenter</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="QGroupBox" name="commandQueueGroup">
     <property name="title">
      <string>Command Queue</string>
     </property>
     <layout class="QVBoxLayout" name="queueLayout">
      <item>
       <widget class="QLabel" name="queueSizeLabel">
        <property name="text">
         <string>Queue Size: 0</string>
        </property>
        <property name="alignment">
         <set>Qt::AlignCenter</set>
        </property>
       </widget>
      </item>
      <item>
       <widget class="QListWidget" name="commandQueueList">
        <property name="minimumSize">
         <size>
          <width>0</width>
          <height>100</height>
         </size>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
